knowage-vue/src/helpers/commons/tableHelpers.ts

```diff
@@ -20,4 +20,24 @@ export function getInputStep(dataType: string) {
     } else {
         return 'any'
     }
+}
+
+export const formatNumber = (column: any, row: any) => {
+    const number = row[column.field]
+    switch (column.format) {
+        case '####,##':
+            return { useGrouping: false, locale: 'it-IT', minFractionDigits: 2, maxFractionDigits: 2 }
+        case '####,###':
+            return { useGrouping: false, locale: 'it-IT', minFractionDigits: 3, maxFractionDigits: 3 }
+        case '#.###,##':
+            return { useGrouping: true, locale: 'it-IT', minFractionDigits: 2, maxFractionDigits: 2 }
+        case '####':
+            return { useGrouping: false, locale: '', minFractionDigits: 0, maxFractionDigits: 0 }
+        case '####.##':
+            return { useGrouping: false, locale: 'en-US', minFractionDigits: 2, maxFractionDigits: 2 }
+        case '#,###.##':
+            return { useGrouping: true, locale: 'en-US', minFractionDigits: 2, maxFractionDigits: 2 }
+        default:
+            return null
+    }
```

Comment:
The possibility to get the number of decimals as present in the task is not available.
The 4# are fixed, after those the number specified can change and be more than just 2 or 3.
I suggest you to use a regex to get the 3 possible groups (1.thousand grouping,2.comma or no comma,3.fractiondigits).
You can avoid changing the locale, the number editor you provided will use the one already set.